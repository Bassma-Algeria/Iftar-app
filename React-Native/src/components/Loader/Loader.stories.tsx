import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {Loader} from './Loader';

const LoaderStories = storiesOf('Loader', module);

LoaderStories.add('Default', () => <Loader />);
LoaderStories.add('Bigger', () => <Loader size={40} />);
LoaderStories.add('Another Color and style', () => (
  <Loader style={{padding: 30}} color="brown" size={50} />
));
