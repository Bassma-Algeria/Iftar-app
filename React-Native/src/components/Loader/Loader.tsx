import React from 'react';
import {ActivityIndicator, StyleProp, ViewStyle} from 'react-native';
import {COLORS} from '../../theme/Colors';

interface Props {
  size?: number;
  color?: keyof typeof COLORS;
  style?: StyleProp<ViewStyle>;
}

const Loader: React.FC<Props> = props => {
  return <ActivityIndicator {...props} />;
};

export {Loader};
