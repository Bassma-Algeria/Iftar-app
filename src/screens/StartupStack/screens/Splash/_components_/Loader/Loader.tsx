import React from 'react';
import {Animated, View} from 'react-native';

// import AnimatedLoader from 'react-native-animated-loader';

import {styles} from '../../Splash.style';
import {useLoaderAnimation} from './_hooks_/useLoaderAnimation';

interface Props {}

const Loader: React.FC<Props> = () => {
  return (
    <View style={styles.loaderContainer}>
      <FirstCircle />
      <SecondCircle />
      <ThirdCircle />
    </View>
  );
};

const FirstCircle: React.FC = () => {
  const {scale} = useLoaderAnimation();

  return <Animated.View style={[styles.loaderPoint, {transform: [{scale}]}]} />;
};

const SecondCircle: React.FC = () => {
  const {scale} = useLoaderAnimation(300);

  return <Animated.View style={[styles.loaderPoint, {transform: [{scale}]}]} />;
};

const ThirdCircle: React.FC = () => {
  const {scale} = useLoaderAnimation(600);

  return <Animated.View style={[styles.loaderPoint, {transform: [{scale}]}]} />;
};

export {Loader};
