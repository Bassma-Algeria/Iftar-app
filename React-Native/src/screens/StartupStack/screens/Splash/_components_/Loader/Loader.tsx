import React from 'react';
// import AnimatedLoader from 'react-native-animated-loader';

import {Loader as BaseLoader} from '../../../../../../components/Loader/Loader';

import {styles} from '../../Splash.style';

interface Props {}

const Loader: React.FC<Props> = () => {
  return <BaseLoader style={styles.loader} color="brown" size={30} />;
};

export {Loader};
