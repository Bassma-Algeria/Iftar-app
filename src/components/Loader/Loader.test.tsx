import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import {Loader} from './Loader';

describe('Loader Component', () => {
  it('should render correctly with the default style', () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the color', () => {
    const tree = renderer.create(<Loader color="brown" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the size', () => {
    const tree = renderer.create(<Loader size={60} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to add custom styles', () => {
    const tree = renderer.create(<Loader style={{margin: 50}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
