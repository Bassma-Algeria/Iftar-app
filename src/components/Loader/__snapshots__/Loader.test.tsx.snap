// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should be able to add custom styles 1`] = `
<ActivityIndicator
  style={
    Object {
      "margin": 50,
    }
  }
/>
`;

exports[`Loader Component should be able to change the color 1`] = `
<ActivityIndicator
  color="#7C4908"
/>
`;

exports[`Loader Component should be able to change the size 1`] = `
<ActivityIndicator
  size={60}
/>
`;

exports[`Loader Component should render correctly with the default style 1`] = `<ActivityIndicator />`;
